src/org/labkey/test/AssayAPITest.java

```diff
@@ -106,6 +111,51 @@ protected void importAssayAndRun(File assayPath, int pipelineCount, String assay
         assertTextPresent(textToCheck);
     }
 
+    @Test
+    public void testGpatAssayOverAPI() throws Exception
+    {
+        String assayName = "testGpatAssay";
+
+        Connection connection = createDefaultConnection(true);
+        GetProtocolCommand getProtocolCommand = new GetProtocolCommand("General");                      // gets a template from the server
+        ProtocolResponse getProtocolResponse = getProtocolCommand.execute(connection, getCurrentContainerPath());
+
+        Protocol newAssayProtocol = getProtocolResponse.getProtocol();
+        newAssayProtocol.setName(assayName)
+            .allowQCStates(true)
+            .setAllowEditableResults(true)
+            .setEditableResults(true)
+            .setEditableRuns(true);
+        SaveProtocolCommand saveProtocolCommand = new SaveProtocolCommand(newAssayProtocol);
+        ProtocolResponse saveProtocolResponse = saveProtocolCommand.execute(connection, getCurrentContainerPath());
+
+        assertTrue(saveProtocolResponse.getProtocol().getAllowQCStates());
+        assertTrue(saveProtocolResponse.getProtocol().getAllowEditableResults());
+        assertTrue(saveProtocolResponse.getProtocol().getEditableResults());
```

Comment:
Consider also doing a separate call to GetProtocolCommand to make sure that the values stick

Reply:
good idea.
Doing so brought up that the GetProtocolCommand and Protocol classes disagree on whether the protocolID should be an int or a long- I've normalized them on 'Long'